SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -500,24 +546,39 @@ void USpatialNetDriver::OnMapLoaded(UWorld* LoadedWorld)
 		return;
 	}
 
-	// If we're the client, we can now ask the server to spawn our controller.
-	if (!IsServer())
+	if (IsServer())
 	{
-		// If we know the GSM is already accepting players, simply spawn.
-		if (GlobalStateManager->bAcceptingPlayers && GetWorld()->RemovePIEPrefix(GlobalStateManager->DeploymentMapURL) == GetWorld()->RemovePIEPrefix(GetWorld()->URL.Map))
+		if (GlobalStateManager != nullptr)
 		{
-			PlayerSpawner->SendPlayerSpawnRequest();
-			bWaitingForAcceptingPlayersToSpawn = false;
+			// Increment the session id, so users don't rejoin the old game.
+			GlobalStateManager->SetCanBeginPlay(true);
+			GlobalStateManager->TriggerBeginPlay();
+			GlobalStateManager->SetAcceptingPlayers(true);
+			GlobalStateManager->IncrementSessionID();
 		}
 		else
 		{
-			checkNoEntry();
+			UE_LOG(LogSpatial, Log, TEXT("Trying to call GlobalStateManager->SetAcceptingPlayers(true) when GlobalStateManager is nullptr"),);
```

Comment:
```suggestion
			UE_LOG(LogSpatial, Error, TEXT("Map loaded on server but GlobalStateManager is not properly initialised. Session cannot start, players cannot connect."));
```
Nit: There was a comma at the end there :D